design-documents/graph-ql/coverage/catalog/remove-product-attributes.md

```diff
@@ -0,0 +1,45 @@
+# Remove product dynamic attributes from ProductInterface
+
+## Current state
+
+Currently, Magento exposes EAV attributes as part of the product schema as top-level fields of ProductInterface. As consequences:
+* The merchant can change the GraphQL from the admin, which makes it hard to cache GraphQL schema.
+* Regular product interface overloaded with fields and almost unreadable.
+* It is quite often practice to create product programmatically, so the number of attributes could be really excessive with tens thousands of attributes.
+* Two Magento instances do not have the same GraphQL schema. As a result, it is near impossible to build a simple client/SDK, which could be easily transferred from one instance to another.
+* It is impossible to aggregate the several Magento instances behind the common [BFF](https://docs.microsoft.com/en-us/azure/architecture/patterns/backends-for-frontends).
+
+## Proposed solution
+
+We should to remove attributes dynamic attributes,
+except system attributes which makes sense for the storefront,
+from the product top level and use a [dynamic container](https://github.com/magento/architecture/blob/master/design-documents/graph-ql/custom-attributes-container.md) instead.
+
+
+```graphql
+## ProductInteface with system EAV attributes 
+type ProductInterface {
+    uid: ID
+    name: String
+    sku: String
+    attribute_set_id: Int
+    description: ComplexTextValue
+    short_description: ComplexTextValue
+    meta_title: String
+    meta_keyword: String
+    meta_description: String
+    created_at: String
+    updated_at: String
+    custom_attributes: [CustomAttribute]
+    ...
+}
+```
+
+## System EAV attributes that should be deprecated at the storefront
+
+* `special_price: Float`  These fields should not be used a the source of the discount since the discount also could be caused by group price or rule price, which are not reflected in this field.
```

Comment:
I think you can use it with out group prices or rule prices. Do you propose to use catalog price rules for simple cases? Seems like a large change, since we will have to deprecate and remove this feature completely from Magento.

Reply:
`special price` that we have here is the admin business logic field and does not represent the special price that we need at the storefront and it can not be used for the scenarios at the storefront.
Also it could be very misleading; the only and the right way to see the discounted price is to check the final and regular prices.

Reply:
catalog rule price, group price, special price all of them admin fields with the only purpose select the least of them and compare to the base price if the resulting price is lower than the product has a special price that makes sense for the storefront, raw EAV attribute itself does not have any sense.